repos/spack_repo/builtin/build_systems/cmake.py

```diff
@@ -406,6 +413,7 @@ def std_args(pkg: PackageBase, generator: Optional[str] = None) -> List[str]:
 
         _conditional_cmake_defaults(pkg, args)
         _maybe_set_python_hints(pkg, args)
+        _dependencies_cmake_extra_args(pkg, args)
```

Comment:
> do you have a better solution here?

Can't we just deal with this in the same way we do for Python above, and code a `_maybe_set_mesa_args`? If we get more cases we can think about generalizing the mechanism. In any case it's probably sensible to stick to just direct deps.